### PR TITLE
Add build pr image and publish stage to build pipeline

### PR DESCRIPTION
**Pull request by Yogeshkumarb57**

### CODE REVIEW

**Review by Yogeshkumarb57 — COMMENTED** [review]
@akarollil Changes reviewed and looks good. You can merge it once approved. I cant approve this PR as github is not allowing me to approve because it was created by me.

**Review by johnson-jeb — APPROVED** [review]

**Pull request merged**
